library/base.go

```diff
@@ -0,0 +1,227 @@
+package library
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+type runningProcess struct {
+	cmd     *exec.Cmd
+	sender  Sender
+	output  *SafeBuffer
+	tracker *StatusTracker
+	done    <-chan bool
+	pid     int
```

Comment:
Do you need to store the PID here if you have a pointer to the `exec.Cmd` already?